### DISCUSSION TITLE
How to feed Networkx with pandas dataframe?

### DISCUSSION THREAD

**Discussion by julianaabdias** [networkx/networkx#5108] [Q&A]
Hi!

I have a dataframe which I'm trying to use to feed Networkx, but I'm not getting the results I expect.

Here is how my dataframe looks originally:

You'll see that id_x and id_y repeats a lot of times, because this dataframe is measure the contacts they have among each other every second during a day. 

          id_x	id_y
9	2225426	2417161
10	2225426	2417161
11	2225426	2417161
13	2225426	2417161
14	2225426	2417161
15	2225426	2417161
16	2225426	2417161
17	2225426	2417161
18	2225426	2417161
19	2225426	2417161
20	2225426	2417161
21	2225426	2417161
22	2225426	2417161
57	2225426	2417161
59	2225426	2417161
71	2225426	2417161
72	2225426	2417161
73	2225426	2417161
99	2225426	2420330
100	2225426	2420330
101	2225426	2420330
102	2225426	2420330
109	2225426	2420330
110	2225426	2420330
111	2225426	2420330
129	2225426	2420330
131	2225426	2420330
141	2225426	2420330
142	2225426	2420330
143	2225426	2420330
144	2225426	2420330
145	2225426	2420330
146	2225426	2420330
147	2225426	2420330
148	2225426	2420330
149	2225426	2420330
150	2225426	2420330
152	2225426	2420330
153	2225426	2420330
154	2225426	2420330
160	2225426	2420330
161	2225426	2420330
162	2225426	2420330
163	2225426	2420330
169	2225426	2420330
170	2225426	2420330
171	2225426	2420330
172	2225426	2420330
176	2225426	2420330
177	2225426	2420330

In order to create an appropriate dataframe to feed the network, I remove the duplicates with the following code:
`df_nodes = df_nodes.drop_duplicates()`

Then I got the following result:
tag_id_x | tag_id_y
2225426 | 2417161
2225426 | 2420330
2225426 | 2420476
2225426 | 2421828
2225426 | 2421869

Now only x is repeating, which I thought was correct because I needed to indicate for every x which y is connected to it.

Problem is, when I feed this dataframe into the network I get the same number of edges for everyone, as everyone was connected to everyone else.

`nx.info(G)`
Name: \nType: Graph\nNumber of nodes: 109\nNumber of edges: 5886\nAverage degree: 108.0000'

When I go back to the dataframe and count the number of y connected to each x, I get a different result:
`df_nodes.groupby(["tag_id_x"])["tag_id_y"].count()`
tag_id_x
2225426    47
2225534    63
2225588    93
2225648    77
2231631    59
           ..
2433119     3
2433145    42
2433148    11
2433150    76
2433164    46
Name: tag_id_y, Length: 108, dtype: int64

I thought it was pretty straightforward to feed the network using the in-built method from_pandas_edglist:
`'edge_list = pd.DataFrame({'from': df_nodes["tag_id_x"], 'to':   df_nodes["tag_id_y"]})`
`G = nx.from_pandas_edgelist(edge_list, source = 'from', target = 'to')`

Am I doing something wrong here?

**Comment by dschult** [comment]
Try a small example where you can see what is going on. There is a lot we can't see from what you are showing us and we can't see your whole problem. So try to do some small tests.  Something like:

```python
In [4]: df= pd.DataFrame({"tag_id_x": [2225426, 2225426, 2225426, 2225426, 2225426], "tag_id_y": [2417161, 2420330, 2420476, 2421828, 2421869]})

In [5]: G=nx.from_pandas_edgelist(df, source="tag_id_x", target="tag_id_y")

In [6]: G.edges
Out[6]: EdgeView([(2225426, 2417161), (2225426, 2420330), (2225426, 2420476), (2225426, 2421828), (2225426, 2421869)])
```
